Loop/Managers/Alerts/AlertStore.swift

```diff
@@ -65,20 +69,46 @@ extension AlertStore {
     }
     
     public func recordAcknowledgement(of identifier: Alert.Identifier, at date: Date = Date(),
-                                      completion: ((Result<Void, Error>) -> Void)? = nil) {
-        recordUpdateOfLatest(of: identifier, with: { $0.acknowledgedDate = date }, completion: completion)
+                                      _ completion: ((Result<Void, Error>) -> Void)? = nil) {
+        recordUpdateOfLatest(of: identifier, field: "acknowledgedDate", with: { $0.acknowledgedDate = date }, completion)
     }
     
     public func recordRetraction(of identifier: Alert.Identifier, at date: Date = Date(),
-                                 completion: ((Result<Void, Error>) -> Void)? = nil) {
-        recordUpdateOfLatest(of: identifier, with: { $0.retractedDate = date }, completion: completion)
+                                 _ completion: ((Result<Void, Error>) -> Void)? = nil) {
+        recordUpdateOfLatest(of: identifier, field: "retractedDate", with: { $0.retractedDate = date }, completion)
     }
     
+    public func lookupAllUnacknowledged(completion: @escaping (Result<[StoredAlert], Error>) -> Void) {
+        managedObjectContext.perform {
+            do {
+                let fetchRequest: NSFetchRequest<StoredAlert> = StoredAlert.fetchRequest()
+                fetchRequest.predicate =  NSCompoundPredicate(andPredicateWithSubpredicates: [
+                    NSPredicate(format: "acknowledgedDate == nil"),
+                    NSPredicate(format: "retractedDate == nil")
+                ])
+                fetchRequest.sortDescriptors = [ NSSortDescriptor(key: "modificationCounter", ascending: true) ]
+                fetchRequest.fetchLimit = 20 // TODO: Seems reasonable to put some limit here, but is this a good value?
+                let result = try self.managedObjectContext.fetch(fetchRequest)
+                completion(.success(result))
+            } catch {
+                completion(.failure(error))
+            }
+        }
+    }
+    
+}
+
+// MARK: Private functions
+
+extension AlertStore {
+    
     private func recordUpdateOfLatest(of identifier: Alert.Identifier,
+                                      field: String,
```

Comment:
[nit] This seems a bit fragile if you get the field name wrong. Instead, consider three functions, `recordUpdateOfLatestWithAcknowledgedDate`, `recordUpdateOfLatestWithRetractedDate`, and `recordUpdateOfLatest`. The latter takes a predicate and the two former pass a predicate based upon the appropriate named property.

Reply:
I'll do something like this